src/test/java/org/broadinstitute/hellbender/engine/ReferenceMemorySourceTest.java

```diff
@@ -0,0 +1,58 @@
+package org.broadinstitute.hellbender.engine;
+
+import com.google.api.client.util.Lists;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Tests for ReferenceMemorySource
+ */
+public class ReferenceMemorySourceTest {
+
+    private final byte[] ref1Bytes = "CATCAT".getBytes();
+    private final SimpleInterval interval1 = new SimpleInterval("chr1", 100, 105);
+    private final ReferenceBases ref1 = new ReferenceBases(ref1Bytes, interval1);
+    private final SimpleInterval interval2 = new SimpleInterval("chr1", 101, 103);
+    private final byte[] ref2Bytes = "ATC".getBytes();
+    private final SimpleInterval interval3 = new SimpleInterval("chr1", 101, 105);
+    private final byte[] ref3Bytes = "ATCAT".getBytes();
+    private final SAMSequenceDictionary seqDir;
+
+    public ReferenceMemorySourceTest() {
+        List<SAMSequenceRecord> l = new ArrayList<>();
+        l.add(new SAMSequenceRecord("chr1",1000000));
+        seqDir = new SAMSequenceDictionary(l);
+    }
```

Comment:
Test classes should not use constructor for initialization -- use TestNG's `@BeforeClass`